packages/core/src/domain/ai-brief/modify-resources.ts

```diff
@@ -0,0 +1,596 @@
+    return getSlimDiagnosticReport(res);
+  }
+
+  if (res.resourceType === "Observation") {
+    return getSlimObservation(res);
+  }
+
+  if (res.resourceType === "Medication") {
+    return getSlimMedication(res);
+  }
+
+  if (res.resourceType === "MedicationRequest") {
+    return getSlimMedicationRequest(res);
+  }
+
+  if (res.resourceType === "MedicationStatement") {
+    return getSlimMedicationStatement(res);
+  }
+
+  if (res.resourceType === "MedicationAdministration") {
+    return getSlimMedicationAdministration(res);
+  }
+
+  if (res.resourceType === "Condition") {
+    return getSlimCondition(res);
+  }
+
+  if (res.resourceType === "Organization") {
+    return getSlimOrganization(res);
+  }
+
+  if (res.resourceType === "Location") {
+    return getSlimLocation(res);
+  }
+
+  return undefined;
+}
+
+export type SlimPatient = Omit<Patient, "name"> & {
+  name?: string | undefined;
+  reference?: Record<string, string>;
+};
+
+export function getSlimPatient(res: Patient): SlimPatient {
+  const updRes = cloneDeep(res);
+  const { address, telecom, text, id, ...otherFields } = updRes;
+  return {
+    ...otherFields,
+    name: getNameString(updRes.name),
+  };
+}
+
+type AllergyContext = {
+  substance: string | undefined;
+  manifestations: string | undefined;
+};
+
+export type SlimAllergyIntolerance = Omit<
+  AllergyIntolerance,
+  "status" | "clinicalStatus" | "reaction"
+> & {
+  status?: string | undefined;
+  context?: AllergyContext[] | undefined;
+  reference?: Record<string, string>;
+};
+
+function getSlimAllergyIntolerance(res: AllergyIntolerance): SlimAllergyIntolerance {
+  const updRes = cloneDeep(res);
+  const status = Array.from(
+    new Set(updRes.clinicalStatus?.coding?.flatMap(coding => coding.code || []))
+  ).join(", ");
+
+  const context = updRes.reaction?.flatMap(reaction => {
+    const substance = getUniqueDisplaysString(reaction.substance);
+    const manifestations = getUniqueDisplaysString(reaction.manifestation);
+    if (!substance && !manifestations) return [];
+    return { substance, manifestations };
+  });
+
+  delete updRes.reaction;
+  delete updRes.clinicalStatus;
+
+  return {
+    ...updRes,
+    status: isUselessStatus(status) ? undefined : status,
+    context,
+  };
+}
+
+export type SlimImmunization = Omit<Immunization, "vaccineCode" | "site" | "route"> & {
+  vaccineCode?: string | undefined;
+  site?: string | undefined;
+  route?: string | undefined;
+  reference?: Record<string, Partial<SlimOrganization>>;
+};
+
+function getSlimImmunization(res: Immunization): SlimImmunization | undefined {
+  const updRes = cloneDeep(res);
+  if (updRes.vaccineCode) {
+    const resVaccineCodeString = JSON.stringify(updRes.vaccineCode).toLowerCase();
+    if (
+      resVaccineCodeString.includes("no data") ||
+      resVaccineCodeString.includes("no immunization")
+    ) {
+      return undefined;
+    }
+  }
+
+  // Remove the unwanted properties directly
```

Comment:
nit: I'd only leave the comment if it was to explain why we're removing these props. The code itself it pretty obvious what it does.